setup.py

```diff
@@ -776,30 +776,8 @@ def find_runtime_path(self, locations=None):
     # Force the 1.8.x HDF5 API even if the library as been compiled to use the
```

Comment:
LGTM, maybe this comment could be updated as well, and I think some ifdefs in `utils.h` may also be removed. Then there's `utilsextension.get_native_type`, but simplifying that may be more involved.

Thanks for the cleanup!

Reply:
Thanks @ivilata 
I have performed some cleanup.
I kindly ask you to give another look if you can.